src/networking.c

```diff
@@ -2353,21 +2355,46 @@ NULL
     } else if (!strcasecmp(c->argv[1]->ptr,"id") && c->argc == 2) {
         /* CLIENT ID */
         addReplyLongLong(c,c->id);
+    } else if (!strcasecmp(c->argv[1]->ptr,"info") && c->argc == 2) {
+        /* CLIENT INFO */
+        sds o = catClientInfoString(sdsempty(), c);
+        o = sdscatlen(o,"\n",1);
+        addReplyVerbatim(c,o,sdslen(o),"txt");
+        sdsfree(o);
     } else if (!strcasecmp(c->argv[1]->ptr,"list")) {
         /* CLIENT LIST */
         int type = -1;
+        sds o = NULL;
         if (c->argc == 4 && !strcasecmp(c->argv[2]->ptr,"type")) {
             type = getClientTypeByName(c->argv[3]->ptr);
             if (type == -1) {
                 addReplyErrorFormat(c,"Unknown client type '%s'",
                     (char*) c->argv[3]->ptr);
                 return;
-             }
+            }
+        } else if (c->argc > 3 && !strcasecmp(c->argv[2]->ptr,"id")) {
+            int j;
+            o = sdsempty();
+            for (j = 3; j < c->argc; j++) {
+                long long cid;
+                if (getLongLongFromObjectOrReply(c, c->argv[j], &cid,
+                            "Invalid client ID")) {
+                    sdsfree(o);
+                    return;
+                }
+                client *cl = lookupClientByID(cid);
+                if (cl) {
+                    o = catClientInfoString(o, cl);
+                    o = sdscatlen(o, "\n", 1);
+                }
```

Comment:
Should we also throw the invalid client ID on the else here instead of silently failing?

Reply:
Considered that, but that would mean failing the entire operation and returning an error instead of a verbatim text. To me that would make sense if we only handle a single ID at a time, but I don't feel too strongly about that.

Reply:
I was on the fence about this, I would be inclined to throw an error, but am fine either way. I would like to have some indication that an ID was requested but nothing was found, like a NULL response, but we can't do that here since it's trying to stay compatible with CLIENT list. It also seems annoying to have the request fail because a client was disconnected for some reason.

Reply:
unlike HMGET etc, where the index of the request has to match the index of the response. this one is a verbatim string, and each line has the ID which the user can match to the ones he requested.
it's not ideal, but considering the alternatives i think that's the right thing to do (current code).